### PR TITLE
use project data from submodule

### PR DESCRIPTION
**Pull request by afeld**
Ha, I didn't realize @mbland had already done this, but I coincidentally created a branch with the same name and repeated the work. Rebased on top of his earlier commit to add instructions.

I'm not sure how the dashboard is currently deployed, but we may need to change some configuration so it pulls in the submodule.

### CODE REVIEW

**Comment by gboone** [comment]
@afeld thanks for this. My experience with submodules is that it's tricky keeping them up to date. Wondering if we can [point the module to a branch instead of a commit sha](https://github.com/18F/dashboard/pull/169/files#diff-7125153abbcb2e949a1b4f81cbe22af5R1) and if that would allow us to get fresh data by running `git submodule update`? 

Also, I think we should include a jekyll command to refresh the data so we can more easily add it to the build flow. I can take that on.

**Comment by mbland** [comment]
@gboone Sounds good to me. Wonder if we could publish such a subcommand implementation as a Gem...or maybe even get it added to Jekyll somehow?

**Comment by afeld** [comment]
Submodules are generally a pain in the :horse: – looking into [some alternatives](https://workflowy.com/s/bTk4ya3RKG).

**Comment by afeld** [comment]
As a temporary (or maybe "good enough") solution, [added a little `update` script](https://github.com/18F/dashboard/commit/56788b9263b2169291a3fc96c592d65853a22d22).

**Comment by gboone** [comment]
I'm new enough to ruby that I still don't fully understand how gems, bundler, and gem publishing work but am eager to learn. We can definitely publish it as a jekyll plugin and probably contribute it back to the project eventually.

**Comment by mbland** [comment]
BTW, I just learned that I could add branch info to hub/.gitmodules like so:

```
[submodule "_data/private"]
        path = _data/private
        url = git@github.com:18F/data-private.git
        branch = master
[submodule "_data/public"]
        path = _data/public
        url = git@github.com:18F/data-public.git
        branch = master
```

Seems to do exactly the right thing, so long as you commit the submodule changes first and run 'git submodule update --remote' in the parent.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "_data"]
+	path = _data
+	url = https://github.com/18F/data-public.git
```
Comment (mbland) [inline .gitmodules:3]: This should also have the line:

```
    branch = master
```

Note that's a tab at the beginning!

**Comment by mbland** [comment]
Not sure the bootstrap and update scripts are necessary; I've updated the README for https://github.com/18F/data-public to spell out how to work with the submodule. Of course, if others find the bootstrap and update scripts helpful, by all means keep them. Comments/questions on the data-public/README are welcome as well.

**Comment by afeld** [comment]
Good tip re: the branch name. Removed the `update` script, but I still think `bootstrap` is useful.

**Comment by gboone** [comment]
653cf07 adds a bit to the fabfile that will make sure the data directory is updated whenever Dashboard deploys. We should figure out how we write a webhook from 18f/data-public to automatically refresh the data directories on 18F/18f.gsa.gov and Dashboard before we roll this out further.

Another thought I had this morning might be following [the advice here](https://esq.io/blog/posts/dogfooding-with-jekyll/) and keeping the data in a directory independent of the repo and then pointing Jekyll to it with the _config.yml. We may not even have to use git modules to keep it updated if we do that.

**Comment by mbland** [comment]
@gboone I'm wary because that advice advocates publishing the data as an endpoint of the site that "owns" it. Any push to that website will effective cause non-hermetic updates to sites that depend on it whenever they're regenerated. The nice property of submodules is that dependent projects will not be affected until someone runs `git submodule update --remote`, and you get a SHA hash identifying the version of the data you're working with.

**Comment by gboone** [comment]
Good point, @mbland, I hadn't thought about that. 

I'll take a crack at writing the hookshot script for 18F/data-public today or tomorrow. I got some ideas about how we might approach it.

**Comment by afeld** [comment]
The approach described in [the blog post](https://esq.io/blog/posts/dogfooding-with-jekyll/) is redundant in this case since that data is already accessible from the separate repository, but I don't feel strongly about it.

Re: the auto-updating of the submodule: it's a good thing, until the schema changes, which would then break the sites that rely on it. A good in-between may be to automatically update the data, but then make sure that the site builds successfully before publishing it. Continuous integration!

**Comment by gboone** [comment]
I like that idea. Maybe we use HTML-proofer in our CI to define a successful build.

**Comment by afeld** [comment]
'jekyll' returning a zero status would be a good start, and running html-proofer for internal links would be a nice-to-have. Running w/ external link checking can cause PR builds to fail for broken links, when that person didn't touch them. I prefer keeping the CI tests idempotent, cutting out the external dependencies.

**Comment by gboone** [comment]
Checking links is part of it, I was thinking more to make sure that all the html we expect to be on the page is actually there. Perhaps selenium tests are better for this. I'm open to ideas. It has to be more than jekyll exiting clean, though, because Jekyll will build successfully without the datafiles present. For example, when I ran `jekyll serve` on Dashboard without having updated the submodule, Jekyll exited `0` and the site built just fine, but all the projects were missing.

**Pull request merged**